File: javascript_work/0310/ex02.js
function Person(name, age){
    this.name = name;
    this.age = age;
}

// Person.prototype.greet = function(){
//     console.log(`hello 나는 ${this.name} 나이는 ${this.age}`);
// }
Person.prototype={
    greet:function(){
        console.log(`hello 나는 ${this.name} 나이는 ${this.age}`);
    }
}

const p1 = new Person("홍길동",20);
const p2 = new Person("박길동",30);

p1.greet();
p2.greet();

console.log(p1.constructor);
console.log(p2.constructor);

File: self_c/ex0416/ex02.c
#define _CRT_SECURE_NOWARNINGS
#include <stdio.h>
#include <string.h>

typedef struct _Person{
    char name[20];
    int age;
    char address[100];
}Person;

int main(){
    Person p1 = {.name="홍길동", .age=25, .address="서울시 용산구 한남동"};
    
    printf("이름 : %s\n",p1.name);
    printf("나이 : %d\n",p1.age);
    printf("주소 : %s\n",p1.address);
    
    Person p2 = {"고길동",30,"서울시 용산구 한남동"};
    printf("이름 : %s\n",p2.name);
    printf("나이 : %d\n",p2.age);
    printf("주소 : %s\n",p2.address);
    
    return 0;
}
